src/v/cloud_roles/request_response_helpers.cc

```diff
@@ -94,57 +115,49 @@ json::Document parse_json_response(iobuf resp) {
     return doc;
 }
 
+static ss::future<api_response> make_post_request(
```

Comment:
it's odd, but my IDE is saying this function isn't used

Reply:
This is strange, perhaps an issue with clangd. In clion I do not see a warning on this or `make_get_request`, but clion uses some extra logic along with clangd. 

I opened the same file in emacs with clangd configured and there I do see this warning along with on `make_get_request`. Although, when I search for references for these functions it does show the usage as expected:

```
src/v/cloud_roles/request_response_helpers.cc
118: static ss::future<api_response> make_post_request(
157:                 return make_post_request(
```

```
src/v/cloud_roles/request_response_helpers.cc
 79: static ss::future<api_response> make_get_request(
104:               return make_get_request(client, req, timeout);
```

Reply:
I tried switching the binary emacs uses for clangd to the one clion uses internally and this warning went away. The one clion uses is newer:

```
abhijat@terminus-est ~/dev/redpanda
 (ntp-archiver-tests-stop-fixture-scheduler) $ ./vbuild/llvm/install/bin/clangd --version
clangd version 14.0.3 (git@github.com:abhijat/redpanda.git 13472a60b1cee71b9e6cc5092e33d345dff141e7)
Features: linux
Platform: x86_64-unknown-linux-gnu
abhijat@terminus-est ~/dev/redpanda
 (ntp-archiver-tests-stop-fixture-scheduler) $ /home/abhijat/.local/share/JetBrains/Toolbox/apps/CLion/ch-0/222.4345.21/bin/clang/linux/clangd --version
clangd version 15.0.0 (/mnt/agent/work/1c7c4e45c2cf5a08/llvm-project/clang cf0e63a7ed8157d9cd0ce6f98d53d3136df35f9f based on LLVM f1df6515e3fe54f688739927c57cb6adec01c307 revision)
Features: linux
Platform: x86_64-unknown-linux-gnu
```

Reply:
wow, thanks for investigating. and good to know it's fixed in a newer version of clang.

Reply:
I think unfortunately this may not be because of clangd, I realised when trying to do a change from emacs today that the clangd supplied with CLion did not work correctly, it crashed whenever I made a change and it did not show any other warnings either. 

So the absence of the warning was most probably due to CLion's clangd not working as a standalone tool.

Reply:
interesting. i'm still very curious about this. it looks like a very basic case it's unclear how clangd would be confused :shrug:

Reply:
i just spent 2 hours on this haha. it appears that clangd has trouble "seeing" through certain layers of calls.